2016-04-28  Eric Jason Chuu  <classech@lnxsrv07.seas.ucla.edu>

	* timespec.h (timespec_cmp): Changed the ints to signed long long
	ints to avoid overflow. Took the difference of these signed long
	long ints and compared them to determine the return.

--- timespec.h	2016-04-28 22:02:58.606579000 -0700
+++ timespec.h	2016-04-28 22:27:49.013448000 -0700
@@ -45,8 +45,23 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  signed long long int a_long = (signed long long int) a.tv_sec;
+  signed long long int b_long = (signed long long int) b.tv_sec;
+  signed long long int diff = a_long - b_long;
+  
+  // -1 if a < b
+  // 0  if a == b
+  // 1  if a > b
+  if (diff < 0)
+    return -1;
+  else if (diff > 0)
+    return 1;
+  else
+    return 0;
+    
+  // OLD IMPLEMENTATION -- CAUSES OVERFLOW WHEN LARGE INTEGER
+  //int diff = a.tv_sec - b.tv_sec;
+  //return diff ? diff : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
